Source/core/CyclicBuffer.cpp

```diff
@@ -353,66 +353,68 @@ namespace Core {
         uint32_t head = 0;
         uint32_t offset = 0;
 
-        while (!foundData) {
-            oldTail = _administration->_tail;
-            head = _administration->_head;
-            offset = oldTail & _administration->_tailIndexMask;
-
-            result = Used(head, offset);
-            if (result == 0) {
-                //no data, no need in trying reading
-                break;
-            }
+        if (length) {
```

Comment:
Think this is the only actual change, correct? (github diff makes that hard to see)
We always explicitly compare with something in our code (even for a bool :) ) so I guess the above could be if(length > 0)?
And I guess this was added for the case somebody would call this with length 0?

Reply:
The indentation indeed marks the actual change.